File: compoundwords/compoundwords.py
# Rating: ~ 1.8 / 10
# Link: https://open.kattis.com/problems/compoundwords
from itertools import combinations
from sys import stdin
def main():
    words = []
    for line in stdin.readlines():
        words += line.strip().split()
    combos = set()
    for a, b in combinations(words, 2):
        combos.add(a + b)
        combos.add(b + a)
    for word in sorted(combos):
        print(word)

if __name__ == "__main__":
  main()

File: compoundwords/compoundwords.cpp
/*
Rating: ~ 1.8 / 10
Link: https://open.kattis.com/problems/compoundwords
*/

#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

void fast() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
}

int main() {
  fast();
  vector<string> words;
  string input;
  while (cin >> input) {
    // input can consist of multiple words space separated
    stringstream ss(input);
    string word;
    while (ss >> word) {
      words.push_back(word);
    }
  }
  set<string> compound;
  for (int i = 0; i < words.size(); i++) {
    for (int j = 0; j < words.size(); j++) {
      if (i != j) {
        compound.insert(words[i] + words[j]);
      }
    }
  }

  for (auto it = compound.begin(); it != compound.end(); it++) {
    cout << *it << endl;
  }
  return 0;
}
